hugo/assets/scripts/ast/ast-tools.ts

```diff
@@ -0,0 +1,152 @@
+
+export class LangiumAST {
```

Comment:
Would be good to leave a brief summary about the purpose/limitation of this class. Something akin to this is used to help with de-serializing & processing Langium AST nodes.